pkg/controller/controller.go

```diff
@@ -37,14 +53,69 @@ func AddToManager(m manager.Manager) error {
 		}
 	}
 	for _, f := range AddToManagerWithActuatorFuncs {
-		// TODO: cleanup hard coded AWS instantiation here:
-		awsActuator, err := awsactuator.NewAWSActuator(m.GetClient(), scheme.Scheme)
+		// Check if this is an AWS cluster and if not, add a dummy actuator:
+		// TODO: Use infrastructure type to determine this in future, it's not being populated yet:
+		// https://github.com/openshift/api/blob/master/config/v1/types_infrastructure.go#L11
+		var err error
+		var a actuator.Actuator
+		isAWS, err := isAWSCluster(m)
 		if err != nil {
-			return err
+			log.Fatal(err)
+		}
+		if isAWS {
+			log.Info("initializing AWS actuator")
+			a, err = awsactuator.NewAWSActuator(m.GetClient(), m.GetScheme())
+			if err != nil {
+				return err
+			}
+		} else {
+			log.Info("initializing no-op actuator (not an AWS cluster)")
+			a = &actuator.DummyActuator{}
 		}
-		if err := f(m, awsActuator); err != nil {
+		if err := f(m, a); err != nil {
 			return err
 		}
 	}
 	return nil
 }
+
+func isAWSCluster(m manager.Manager) (bool, error) {
+	client, err := getClient()
+	if err != nil {
+		return false, err
+	}
+	cfgMapName := types.NamespacedName{Name: "cluster-config-v1", Namespace: "kube-system"}
+	installCfgMap := &corev1.ConfigMap{}
+	err = client.Get(context.Background(), cfgMapName, installCfgMap)
+	if err != nil {
+		return false, err
+	}
+	data, ok := installCfgMap.Data["install-config"]
+	if !ok {
+		return false, fmt.Errorf("cluster-config-v1 ConfigMap did not contain install-config data")
+	}
+	yamlStr := string(data)
+	log.Warn(yamlStr)
+	match, err := regexp.MatchString(`.*platform:\s*aws:.*`, yamlStr)
```

Comment:
This feels fragile. Why not [deserialize it](https://github.com/openshift/master-dns-operator/blob/master/pkg/operator/masterdns/operator.go#L490-L500), or even better [use the infrastructure config](https://github.com/openshift/cluster-ingress-operator/blob/master/cmd/cluster-ingress-operator/main.go#L52-L56)

Reply:
I didn't want to deserialize as it introduces a circular dependency, we'd pull the installer and the installer will soon be pulling this project for checking creds.

It's also very temporary, infrastructure config is the long term solution but it's not being populated yet, will eventually switch to that. (there should be a TODO in here somewhere I thought, I will check)